src/alda/parser.clj

```diff
@@ -20,16 +20,12 @@
           :slur              (constantly :slur)
           :flat              (constantly :flat)
           :sharp             (constantly :sharp)
+          :natural           (constantly :natural)
           :dots              #(hash-map :dots (count %))
           :note-length       #(list* 'alda.lisp/note-length %&)
           :duration          #(list* 'alda.lisp/duration %&)
-          :pitch             (fn [s]
-                               (list* 'alda.lisp/pitch
-                                      (keyword (str (first s)))
-                                      (map #(case %
-                                              \- :flat
-                                              \+ :sharp)
-                                           (rest s))))
+          :pitch             #(list* 'alda.lisp/pitch %&)
+          :pitch-notation    #(list* 'alda.lisp/parse-pitch %)
```

Comment:
Then, (following the change I proposed above in `alda.bnf`), you could get rid of the `:pitch-notation` transformation here (since that node will no longer explicitly exist), and have the `:pitch` transformation be `#(list* 'alda.lisp/pitch (apply parse-pitch %&))`